extensions/guacamole-auth-jdbc/modules/guacamole-auth-jdbc-base/src/main/java/org/apache/guacamole/auth/jdbc/tunnel/AbstractGuacamoleTunnelService.java

```diff
@@ -416,6 +462,21 @@ private GuacamoleTunnel assignGuacamoleTunnel(ActiveConnectionRecord activeConne
             GuacamoleClientInformation info, Map<String, String> tokens,
             boolean interceptErrors) throws GuacamoleException {
 
+        // Set up JDBC-specific tokens
+        tokens.put(JDBC_DATE_TOKEN,
+                new SimpleDateFormat(JDBC_DATE_TOKEN_FORMAT)
+                        .format(activeConnection.getStartDate()));
+        tokens.put(JDBC_TIME_TOKEN,
+                new SimpleDateFormat(JDBC_TIME_TOKEN_FORMAT)
+                        .format(activeConnection.getStartDate()));
+        tokens.put(JDBC_CONNECTION_NAME_TOKEN, activeConnection.getConnectionName());
+        tokens.put(JDBC_CONNECTION_ID_TOKEN, activeConnection.getConnectionIdentifier());
+        tokens.put(JDBC_CONNECTION_HOSTNAME_TOKEN,
+                activeConnection.getConnection().getConfiguration().getParameter(JDBC_CONNECTION_HOSTNAME_TOKEN_PARAMTER));
+        tokens.put(JDBC_CONNECTION_PROTOCOL_TOKEN,
+                activeConnection.getConnection().getConfiguration().getProtocol());
```

Comment:
Within `AbstractGuacamoleTunnelService`, I think any additional tokens should be added alongside where we are already adding `HISTORY_UUID`:

https://github.com/apache/guacamole-client/blob/165bd413c0ab9c55b0c0e3614fcb829ef5f8ca95/extensions/guacamole-auth-jdbc/modules/guacamole-auth-jdbc-base/src/main/java/org/apache/guacamole/auth/jdbc/tunnel/AbstractGuacamoleTunnelService.java#L463-L465

Reply:
Okay, I can relocate things.

Reply:
Relocated via rebase.